src/include/stir/ProjDataGEHDF5.h

```diff
@@ -0,0 +1,102 @@
+//
+//
+/*!
+
+  \file
+  \ingroup projdata
+  \ingroup GE
+  \brief Declaration of class stir::GE::RDF_HDF5::ProjDataFromGEHDF5
+
+  \author Palak Wadhwa
+  \author Nikos Efthimiou
+  \author Kris Thielemans
+
+
+*/
+/*  Copyright (C) 2018-2019 University College London
+    Copyright (C) 2018-2019 University of Leeds
+    Copyright (C) 2018-2019 University of Hull
+    This file is part of STIR.
+
+    This file is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 2.1 of the License, or
+    (at your option) any later version.
+
+    This file is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    See STIR/LICENSE.txt for details
+*/
+#ifndef __ProjDataGEDF5_H__
+#define __ProjDataGEDF5_H__
+
+#include "stir/ProjData.h"
+#include "stir/IO/GEHDF5Wrapper.h"
+#include "stir/Array.h"
+
+START_NAMESPACE_STIR
+
+namespace GE {
+namespace RDF_HDF5 {
+
+
+/*!
+  \ingroup projdata
+  \ingroup GE
+  \brief A class which reads projection data from a GE HDF5
+  sinogram file.
+*/
+class ProjDataGEHDF5 : public ProjData
+{
+public:
+
+    static ProjDataGEHDF5* ask_parameters(const bool on_disk = true);
+
+    explicit ProjDataGEHDF5(shared_ptr<const ProjDataInfo> input_proj_data_info_sptr,
+                              const std::string& input_filename);
+
+    explicit ProjDataGEHDF5(shared_ptr<const ProjDataInfo> input_proj_data_info_sptr,
+                              shared_ptr<GEHDF5Wrapper> input_hdf5_sptr);
+
+    explicit ProjDataGEHDF5
+      (shared_ptr<const ExamInfo> input_exam_info_sptr,
+                              shared_ptr<const ProjDataInfo> input_proj_data_info_sptr,
+                              shared_ptr<GEHDF5Wrapper> input_hdf5);
```

Comment:
These constructors take shared pointers for all arguments and then create deep copies of them to ensure that the original object isn't changed elsewhere. It looks like there's no way of copying the `GEHDF5Wrapper`, so that approach can't be taken there. However, it looks like it's opened as read only, so perhaps this isn't a problem. I'm not sure.

Reply:
This mimics`ProjDataFromStream` which takes ` shared_ptr<iostream>`.
https://github.com/UCL/STIR/blob/7dc238f5ec228554f594f1f2228450eb4618472b/src/include/stir/ProjDataFromStream.h#L85-L87
You cannot copy that either. I guess the class interface tells people they have to be careful...

By the way, I'm no sure if we really need these constructors, but will leave them in.